pulpcore/app/viewsets/custom_filters.py

```diff
@@ -356,11 +356,11 @@ def filter(self, qs, value):
             versions_distributions[dist["publication__repository_version"]].append(dist["pk"])
 
         for dist in qs.exclude(repository_version=None).values("repository_version", "pk"):
```

Comment:
```suggestion
        for dist in qs.exclude(repository_version=None).values("repository_version", "pulp_type"):
```
We need to pull in the `pulp_type` or else the `detail_model` call will trigger a db lookup. `pk` is always pulled in, so no need to list it.